File: Distinct Difference[19-Mar-2023]/Solution.java
// 				>> JAVA CODE <<

class Solution {
    public static ArrayList<Integer> getDistinctDifference(int N, int[] A) {
        
        ArrayList<Integer> ans=new ArrayList<>();
        
        HashSet<Integer> set=new HashSet<>();
        HashMap<Integer,Integer> map=new HashMap<>();
        for(var in:A) map.put(in,map.getOrDefault(in,0)+1);
        
        for(int i=0;i<A.length;i++)
        {
            map.put(A[i],map.get(A[i])-1);
            if(map.get(A[i])==0) map.remove(A[i]);
            
            int left=set.size();
            int right=map.size();
            
            ans.add(left-right);
            
            set.add(A[i]);
        }
        
        
            return ans;
    }
}
        

File: Distinct Difference[19-Mar-2023]/Solution.cpp
//			>> C++ CODE <<

vector<int> getDistinctDifference(int N,vector<int>& A) {
    vector<int> ans;
    unordered_set<int> s;
    unordered_map<int, int> m;
    for (int i = 0; i < N; i++) m[A[i]]++;
    
    for (int i = 0; i < N; i++) {
        m[A[i]]--;
        if (m[A[i]] == 0) m.erase(A[i]);
        
        int left = s.size();
        int right = m.size();
        
        ans.push_back(left - right);
        
        s.insert(A[i]);
    }
    return ans;
}